src/UTF8Util.hpp

```diff
@@ -75,25 +75,7 @@ class OPENCC_EXPORT UTF8Util {
   * Returns the length in byte for the previous UTF8 character.
   */
   static size_t PrevCharLength(const char* str) {
-    {
-      const size_t length = NextCharLengthNoException(str - 3);
-      if (length == 3) {
-        return length;
-      }
-    }
-    {
-      const size_t length = NextCharLengthNoException(str - 1);
-      if (length == 1) {
-        return length;
-      }
-    }
-    {
-      const size_t length = NextCharLengthNoException(str - 2);
-      if (length == 2) {
-        return length;
-      }
-    }
-    for (size_t i = 4; i <= 6; i++) {
+    for (size_t i = 1; i <= 6; i++) {
```

Comment:
This doesn't actually prevent errors, if there is any. The change is only about order of tested lengths.